src/lib/forward.jl

```diff
@@ -137,8 +137,28 @@ end
 
 # Use this to allow second derivatives -- this is forward-over-forward,
 # see  https://github.com/FluxML/Zygote.jl/issues/769  for a forward-over-reverse proposal
-@adjoint ForwardDiff.gradient(f, x) = pullback(forwarddiff, x -> ForwardDiff.gradient(f, x), x)
-@adjoint ForwardDiff.jacobian(f, x) = pullback(forwarddiff, x -> ForwardDiff.jacobian(f, x), x)
+@adjoint function ForwardDiff.gradient(f, x)
+  F = typeof(f)
+  Base.issingletontype(F) || @warn """`ForwardDiff.gradient(f, x)` within Zygote cannot track gradients with respect to `f`
+  typeof(f) = $F is not a singleton type""" # maxlog=1 _id=hash(F)
+  pullback(forwarddiff, x -> ForwardDiff.gradient(f, x), x)
+end
+
+@adjoint function ForwardDiff.jacobian(f::F, x) where F
+  Base.issingletontype(F) || @warn """`ForwardDiff.jacobian(f, x)` within Zygote cannot track gradients with respect to `f`
+  typeof(f) = $F is not a singleton type""" # maxlog=1 _id=hash(F)
```

Comment:
Is there a wording that cuts down on the jargon here? e.g.
```suggestion
  Base.issingletontype(F) || @warn """`ForwardDiff.jacobian(f, x)` within Zygote cannot track gradients with respect to `f`
  typeof(f) = $F may be a closure or a callable type with fields""" # maxlog=1 _id=hash(F)
```

Reply:
Didn't think hard at all. Might be nice to mention `issingletontype` so people know precisely what the test is.